plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentmodel/ContentTypeResourceController.java

```diff
@@ -0,0 +1,232 @@
+import org.entando.entando.web.common.exceptions.*;
+import org.entando.entando.web.common.model.*;
+import org.entando.entando.web.userprofile.validator.ProfileTypeValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.net.*;
+import java.util.*;
+
+@RestController
+public class ContentTypeResourceController implements ContentTypeResource {
+
+    private static final String CONTENT_TYPE_CODE = "contentTypeCode";
+    private static final String ATTRIBUTE_CODE = "attributeCode";
+    private static final String MOVEMENT = "movement";
+
+    private final ContentTypeService service;
+    private final ContentTypeValidator validator;
+
+    @Autowired
+    public ContentTypeResourceController(ContentTypeService service, ContentTypeValidator validator) {
+        this.service = service;
+        this.validator = validator;
+    }
+
+    @Override
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    public ResponseEntity<ContentTypeDto> create(
+            @Valid @RequestBody ContentTypeDtoRequest contentType,
+            BindingResult bindingResult)
+            throws URISyntaxException {
+
+        validator.validate(contentType, bindingResult);
+        if(bindingResult.hasErrors()) {
+            throw new ValidationGenericException(bindingResult);
+        }
+        ContentTypeDto result = service.create(contentType, bindingResult);
+
+        return ResponseEntity.created(
+                new URI("/plugins/contentTypes/" + result.getCode()))
+                .body(result);
+    }
+
+    @Override
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    public ResponseEntity<Void> delete(@PathVariable("code") String code) {
+
+        service.delete(code);
+        return ResponseEntity.ok().build();
+    }
+
+    @Override
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    public ResponseEntity<PagedRestResponse<EntityTypeShortDto>> list(RestListRequest listRequest) {
+        validator.validateRestListRequest(listRequest, ContentTypeDto.class);
+        PagedMetadata<EntityTypeShortDto> result = service.findMany(listRequest);
+        validator.validateRestListResult(listRequest, result);
+        return ResponseEntity.ok(new PagedRestResponse<>(result));
+    }
+
+    @Override
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    public ResponseEntity<ContentTypeDto> get(
+            @PathVariable("code") String code) {
+
+        Optional<ContentTypeDto> maybeContentType = service.findOne(code);
+
+        return maybeContentType.map(ResponseEntity::ok)
+                               .orElse(ResponseEntity.notFound().build());
+    }
+
+    @Override
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    public ResponseEntity<ContentTypeDto> update(
```

Comment:
As for the previous comment, maybe `ResponseEntity<SimpleRestResponse<ContentTypeDto>>`

Reply:
Fixed